src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -159,12 +159,26 @@ public void Stop()
                     }
                     catch (Exception ex)
                     {
-                        Logger.Error($"{requestType.MethodName} : {ex}");
-                        await requestContext.SendError(ex.Message);
+                        string exceptionMessage = FormatException(ex);
+                        Logger.Error($"{requestType.MethodName} : {exceptionMessage}");
+                        await requestContext.SendError(exceptionMessage);
                     }
                 });
         }
 
+        private string FormatException(Exception e)
+        {
+            string res = string.Empty;
```

Comment:
you can add the messages to an array and then do a join so that you don't need to do the TrimEnd.

also, we probably should also add a space character between the messages since new line is not going to be displayed in the error dialog.

Reply:
String.Join()

Reply:
updated
![image](https://user-images.githubusercontent.com/21186993/223864902-42afa276-bf00-437a-b02d-80f93c278d05.png)